components/router/router.js

```diff
@@ -4,59 +4,46 @@
  *--------------------------------------------------------------------------------------------*/
 
 import components from 'coherent-gameface-components';
+import { HashHistory, BrowserHistory } from './history';
+import { Route } from './route';
 
 const WILDCARD = '**';
 const HOME = '/';
 
 class Router {
+    /**
+     * @typedef {Record<string, string|Router>} Routes
```

Comment:
Where do we use this type Record? A little confused here, can you elaborate, please?

Reply:
Sure I will! I defined this type because I added a JSDoc to the Router constructor where the `routes` type was `any`. This type definition is defining the `routes` type passed to the constructor. you can check this line - https://github.com/CoherentLabs/GameUIComponents/blob/c6a94429fd9b14c6e73a913247bf7143021804d7/components/router/router.js#L19

Record is used to define the following type - {[key:type]:type}. In the following case it is defining {[key:string]: string| Router}
If fact Record is wrong, it should be Object<string, string|Router> that will define the same but with JSDocs, so I will fix this...